mreg_cli/api/models.py

```diff
@@ -255,42 +262,134 @@ def is_delegated(self) -> bool:
+        repr=False,
+    )
+    """Constructed datetime field from `create_date` in the API.
+
+    WARNING
+    ----
+    DO NOT USE THIS FIELD FOR TIMEZONE-AWARE COMPARISONS!
+    Always use `created_at` instead when comparing with timezone-aware
+    fields such as `update_time`."""
+    updated_at: datetime
     name: str
-    labels: list[int]
+    description: str
 
-    @field_validator("created_at", mode="before")
+    @field_validator("created_at_tz_naive", mode="before")
     @classmethod
-    def validate_created_at(cls, value: str) -> datetime:
-        """Validate and convert the created_at field to datetime.
+    def validate_created_at(cls, value: Any) -> datetime:
+        """Convert a datetime string to a datetime object.
 
-        :param value: Input date string from the JSON.
-        :returns: Converted datetime object.
+        :param value: The input value - should be a datetime string.
+        :returns: The input value converted to a datetime object.
         """
-        return datetime.fromisoformat(value)
+        # Fast path for str (most likely input type)
+        if isinstance(value, str):
+            return datetime.fromisoformat(value)
+        elif isinstance(value, datetime):
+            return value
+        elif isinstance(value, date):
+            return datetime.combine(value, datetime.min.time())
+        return value  # let pydantic throw the ValidationError
+
+    @computed_field
+    def created_at(self) -> datetime:
+        """Creation time."""
+        return self.created_at_tz_naive.replace(tzinfo=self.updated_at.tzinfo)
+
+    @classmethod
+    def get_by_name(cls, name: str) -> Atom | Role:
+        """Get an Atom or Role by name.
+
+        :param name: The name to search for.
+        :returns: The Atom or Role if found.
+        :raises CliWarning: If the Atom or Role is not found.
+        """
+        role_or_atom: Role | Atom | None = None
+        for func in [Atom.get_by_name, Role.get_by_name]:
+            try:
+                role_or_atom = func(name)
+            except Exception:
+                pass
+        if role_or_atom is None:
+            cli_warning(f"Could not find an atom or a role with name: {name!r}")
+        return role_or_atom
+
+    def output_timestamps(self, padding: int = 14) -> None:
+        """Output the created and updated timestamps to the console."""
+        output_manager = OutputManager()
+        output_manager.add_line(f"{'Created:':<{padding}}{self.created_at:%c}")
+        output_manager.add_line(f"{'Updated:':<{padding}}{self.updated_at:%c}")
+
+    def output(self, padding: int = 14) -> None:
+        """Output the host policy object to the console.
+
+        Subclasses should provide their own output method and call this method
+        first to output the commmon fields.
+        """
+        output_manager = OutputManager()
+        output_manager.add_line(f"{'Name:':<{padding}}{self.name}")
+        self.output_timestamps(padding=padding)
```

Comment:
This will now show `Updated:` when doing `policy info`, which it previously didn't. Remove or keep?

Reply:
I did the same for other models, I think it's nice to have. I'm thinking we can add some config options to allow users to customise some aspects of the output as well.

Reply:
Then we definitely need to unify the two `output_timestamps` methods, since I just copied it over to the new `HostPolicy` base class.

Reply:
I look forward to the day when a bit of copy and paste are the problems we fret over. :)